packages/core/src/internal/data-grid/render/data-grid-render.header.ts

```diff
@@ -77,7 +77,7 @@ export function drawGridHeaders(
             ctx.font = theme.baseFontFull;
         }
         const selected = selection.columns.hasIndex(c.sourceIndex);
-        const noHover = dragAndDropState !== undefined || isResizing;
+        const noHover = dragAndDropState !== undefined || isResizing || c.headerRowMarkerDisabled === true;
```

Comment:
Same as the comment above. Not sure if this is event required since the exising logic of `headerRowMarkerDisabled` should already prevent the hover checkbox.

Reply:
Only hides but doesn't prevent interaction; see comment above.